src/main/java/org/takes/rs/RsWithStatus.java

```diff
@@ -117,17 +119,20 @@ public InputStream body() throws IOException {
                 )
             );
         }
-        final Collection<String> head = new LinkedList<String>();
-        head.add(String.format("HTTP/1.1 %d %s", status, reason));
-        boolean first = true;
-        for (final String hdr : origin.head()) {
-            if (first) {
-                first = false;
-            } else {
-                head.add(hdr);
-            }
-        }
-        return head;
+        return new Concat<String>(
+            Collections.singletonList(
+                String.format("HTTP/1.1 %d %s", status, reason)
+            ),
+            new Select<String>(
+                origin.head(),
+                new Condition<String>() {
+                    @Override
+                    public boolean fits(final String element) {
+                        return !RsWithStatus.FIRST.matcher(element).matches();
```

Comment:
@dmzaytsev eghh... I don't like. you're checking every single line for a regex matching.. this is 1) slow and 2) redundant. all you need to do is to filter out the first line. I would use an `AtomicBoolean` flag for that